libmamba/src/core/activation.cpp

```diff
@@ -1215,4 +1215,97 @@ namespace mamba
 
         return out.str();
     }
+
+
+    std::string NuActivator::shell_extension()
+    {
+        return ".nu";
+    }
+
+    std::string NuActivator::shell()
+    {
+        return "nu";
+    }
+
+    std::string NuActivator::hook_preamble()
+    {
+        return "";
+    }
+
+    std::string NuActivator::hook_postamble()
+    {
+        return "";
+    }
+
+    fs::u8path NuActivator::hook_source_path()
+    {
+        return Context::instance().prefix_params.root_prefix / "etc" / "profile.d" / "micromamba.nu";
+    }
+
+    std::pair<std::string, std::string>
+    NuActivator::update_prompt(const std::string& /*conda_prompt_modifier*/)
+    {
+        return { "", "" };
+    }
+
+    std::string NuActivator::script(const EnvironmentTransform& env_transform)
+    {
+        std::stringstream out;
+        auto convert_to_nulist = [](std::string command)
+        {
+            /*Checks shell variable being set and converts shell lists to nushell lists if
+             possible otherwise returns the command as a string*/
+            std::string new_command = command;
+            std::size_t idx = 0;
+            bool add = false;
+            while ((idx = new_command.find(":", idx)) != std::string::npos)
+            {
+                new_command.replace(idx, 1, " ");
+                add = true;
+                idx++;
+            }
+            if (add)
+            {
+                new_command = fmt::format("[{}]", new_command);
+            }
+            return new_command;
+        };
+
+        if (!env_transform.export_path.empty())
+        {
+            out << "$env.PATH = " << fmt::format("[{}]", env_transform.export_path) << "\n";
+        }
+
+        for (const fs::u8path& ds : env_transform.deactivate_scripts)
+        {
+            out << "source " << ds << "\n";
+        }
+
+        for (const std::string& uvar : env_transform.unset_vars)
+        {
+            out << "hide-env " << uvar << "\n";
+        }
+
+        for (const auto& [skey, svar] : env_transform.set_vars)
+        {
+            out << "let " << skey << " = " << svar << "\n";
+        }
+
+        for (const auto& [ekey, evar] : env_transform.export_vars)
+        {
+            // escape parenthesis if present otherwise checks if @evar can be converted to nushell
```

Comment:
Rather than commenting on _what_ is being done here (I can read that much myself!) it would be very helpful to have an explanation of _why_ we are doing it.

Why are we quoting `evar` if and only if it contains a `(`?
Why are we converting it to something else otherwise?

Reply:
In nushell parenthesis equates to evaluating the inside bits. This causes a problem when the environment problem is being set as it thinks ($conda_prompt) is an executable, hence the need to escape this variable inside a nushell. It behaves the same as other shell in other case 
![image](https://github.com/mamba-org/mamba/assets/19485143/30e31c1b-2574-4087-8859-c046562178d5)